Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -362,6 +362,20 @@ inline HRESULT GetTextRenderingParams(IDWriteRenderingParams* originalParams, ID
         return S_OK;
     }
 
+    inline HRESULT Flush() {
+        // We cannot use use ID2D1::Flush here, mainly due to:
+        // 1. Flush flushes the D2D batch, but does not flush the D3D batch, we do not have a way to flush
+        // the D3D batch for WicBitmapRenderTarget. (Not: During the EndDraw procedure this occurs.)
+        // 2. Sometimes a shadow copy of our buffer can be used to do the actual rendering, Flush does not force the copy
+        // over of the shadow copy into the original buffer (EndDraw does)
+        // Thus the best way is to implement Flush is through EndDraw
+        if (_beginEndDrawDepth > 0) {
+            RETURN_IF_FAILED(PopEndDraw(true));
```

Comment:
but that can happen any time...
This is to be generic.

---
In reply to: [106513837](https://github.com/Microsoft/WinObjC/pull/2208#discussion_r106513837) [](ancestors = 106513837)